rollup/sync_service/sync_service.go

```diff
@@ -199,8 +211,8 @@ func (s *SyncService) fetchMessages() {
 			}
 			return
 		case <-t.C:
-			progress := 100 * float64(s.latestProcessedBlock) / float64(latestConfirmed)
-			log.Info("Syncing L1 messages", "processed", s.latestProcessedBlock, "confirmed", latestConfirmed, "collected", numMsgsCollected, "progress(%)", progress)
+			progress := 100 * float64(s.latestProcessedBlock.Load()) / float64(latestConfirmed)
```

Comment:
we should load `s.latestProcessedBlock`just once to a local variable at the beginning of `fetchMessages`, so that we use a consistent value in the entire function. If not, a call to `ResetStartSyncHeight` might change it in the middle of `fetchMessages` and we might use different versions in different calculations.